osgp/protocol-adapter-dlms/osgp-dlms/src/main/java/org/opensmartgridplatform/dlms/objectconfig/dlmsClasses/ProfileGeneric.java

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.opensmartgridplatform.dlms.objectconfig.dlmsClasses;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.opensmartgridplatform.dlms.exceptions.ObjectConfigException;
+import org.opensmartgridplatform.dlms.objectconfig.Attribute;
+import org.opensmartgridplatform.dlms.objectconfig.CaptureObject;
+import org.opensmartgridplatform.dlms.objectconfig.CosemObject;
+import org.opensmartgridplatform.dlms.objectconfig.MeterType;
+import org.opensmartgridplatform.dlms.objectconfig.ObjectProperty;
+import org.opensmartgridplatform.dlms.services.ObjectConfigService;
+
+@Getter
+@NoArgsConstructor
+public class ProfileGeneric extends CosemObject {
+
+  public ProfileGeneric(
+      final String tag,
+      final String description,
+      final int classId,
+      final int version,
+      final String obis,
+      final String group,
+      final String note,
+      final List<MeterType> meterTypes,
+      final Map<ObjectProperty, Object> properties,
+      final List<Attribute> attributes) {
+    super(
+        tag, description, classId, version, obis, group, note, meterTypes, properties, attributes);
+  }
+
+  public List<CaptureObject> getCaptureObjects(
+      final ObjectConfigService service,
+      final String protocol,
+      final String version,
+      final Integer channel)
+      throws ObjectConfigException {
+    final List<CaptureObject> captureObjects = new ArrayList<>();
+
+    final List<Integer> channels;
+    if (this.hasWildcardChannel()) {
+      // If the profile has an x for the channel in the obis code, then return all capture objects
```

Comment:
Can you explain this to me?

Reply:
For some protocol versions, the profile generic can be defined for each channel. In the config, such a profile has a wildcard channel (x), otherwise we need to define the profile multiple times (once for each channel).
If the profile generic is defined for each channel in a protocol version, then that also means that the captured values in the profile are for one channel only: the same channel as the profile.
On the other hand, if the profile generic is defined without channel in a protocol version, then the captured values are for all channels. For example, a combined profile in DSMR4.2.2 has captured values for the M-Bus master value for channels 1 to 4.